include/cantera/thermo/MixtureFugacityTP.h

```diff
@@ -1,577 +1,572 @@
+     *        where in parameters space we are
+     *
+     *  There are three values:
+     *  - WATER_GAS   below the critical temperature but below the critical density
+     *  - WATER_LIQUID  below the critical temperature but above the critical density
+     *  - WATER_SUPERCRIT   above the critical temperature
+     */
+    int phaseState(bool checkState = false) const;
+
+    //! Return the value of the density at the liquid spinodal point (on the
+    //! liquid side) for the current temperature.
+    /*!
+     * @returns the density with units of kg m-3
+     */
+    virtual doublereal densSpinodalLiquid() const;
+
+    //! Return the value of the density at the gas spinodal point (on the gas
+    //! side) for the current temperature.
+    /*!
+     * @returns the density with units of kg m-3
+     */
+    virtual doublereal densSpinodalGas() const;
+
+public:
+    //! Calculate the saturation pressure at the current mixture content for the
+    //! given temperature
+    /*!
+     * @param TKelvin         (input) Temperature (Kelvin)
+     * @param molarVolGas     (return) Molar volume of the gas
+     * @param molarVolLiquid  (return) Molar volume of the liquid
+     * @returns the saturation pressure at the given temperature
+     */
+    doublereal calculatePsat(doublereal TKelvin, doublereal& molarVolGas,
+                             doublereal& molarVolLiquid);
+
+public:
+    //! Calculate the saturation pressure at the current mixture content for the
+    //! given temperature
+    /*!
+     * @param TKelvin   Temperature (Kelvin)
+     * @return          The saturation pressure at the given temperature
+     */
+    virtual doublereal satPressure(doublereal TKelvin);
+	virtual void setToEquilState(const doublereal* lambda_RT);
+    virtual void getActivityConcentrations(double* c) const;
+
+protected:
+    //! Calculate the pressure given the temperature and the molar volume
+    /*!
+     * @param   TKelvin   temperature in kelvin
+     * @param   molarVol  molar volume ( m3/kmol)
+     * @returns the pressure.
+     */
+    virtual doublereal pressureCalc(doublereal TKelvin, doublereal molarVol) const;
+
+    //! Calculate the pressure and the pressure derivative given the temperature
+    //! and the molar volume
+    /*!
+     *  Temperature and mole number are held constant
+     *
+     * @param   TKelvin   temperature in kelvin
+     * @param   molarVol  molar volume ( m3/kmol)
+     * @param   presCalc  Returns the pressure.
+     * @returns the derivative of the pressure wrt the molar volume
+     */
+    virtual doublereal dpdVCalc(doublereal TKelvin, doublereal molarVol, doublereal& presCalc) const;
+
+    virtual void updateMixingExpressions();
+
+
+    //! Solve the cubic equation of state
+    /*!
+     * 
+     * Returns the number of solutions found. If it only finds the liquid
+     * branch solution, it will return a -1 or a -2 instead of 1 or 2.  If it

```

Comment:
I'm not totally sure what this means. What is the alternative when the solution is positive, a gas phase solution? What is the physical meaning of having 2 roots for the equation? Which one do you pick and why? Is this something that a user would be likely to know? Is this function only for developers (I assume not since it is `public`?

Also, a really small nitpick - one space after a period please, just for consistency with the rest of the docs (looking at you @decaluwe :smiley:)

Reply:
The two roots here represent two distinct roots (and the third one is repeated). The -1 and -2 are just the conventions used to represent the liquid phase. These values are further used in child classes (R-K eos and P-R eos) to set the flag foundLiq = True.

Reply:
I guess it would help to clarify then that the positive numbers represent solutions in the gas phase. That is implied by the sentence marked here but I think it would be better to be explicit.

Reply:
Modified the comment to be more clear and explicit.